### PR TITLE
Allow construction of `DeviceBuffer` with a `stream`

### PR DESCRIPTION
**Pull request by jakirkham**
Adds an optional `stream` argument to `DeviceBuffer`'s constructor to allow creating the `rmm::device_buffer` on a particular `stream`.

### CODE REVIEW

File: python/rmm/_lib/device_buffer.pyx [hunk 1]
```diff
@@ -9,18 +9,25 @@ from rmm._lib.lib cimport (cudaError_t, cudaSuccess,
...
             size = 0
 
+        if stream is None:
+            stream = 0
+
         cdef void * data
         if ptr is None:
-            self.c_obj.reset(new device_buffer(<size_t>size))
+            self.c_obj.reset(
+                new device_buffer(<size_t>size, <cudaStream_t>stream)
```
Comment (kkraus14) [inline python/rmm/_lib/device_buffer.pyx]: I think we need a doublecast if someone passes us a python integer here.
```suggestion
                new device_buffer(<size_t>size, <cudaStream_t><uintptr_t>stream)
```

File: python/rmm/_lib/device_buffer.pyx [hunk 2]
```diff
@@ -9,18 +9,25 @@ from rmm._lib.lib cimport (cudaError_t, cudaSuccess,
...
+            )
         else:
             data = <void *> <uintptr_t> ptr
-            self.c_obj.reset(new device_buffer(data, size))
+            self.c_obj.reset(
+                new device_buffer(data, <size_t>size, <cudaStream_t>stream)
```
Comment (kkraus14) [inline python/rmm/_lib/device_buffer.pyx]:
```suggestion
                new device_buffer(data, <size_t>size, <cudaStream_t><uintptr_t>stream)
```

File: python/rmm/_lib/device_buffer.pyx (hunk 1 above)
Comment (jakirkham) [reply python/rmm/_lib/device_buffer.pyx]: Good catch. Thanks!

**Comment by jakirkham** [comment]
Thanks @kkraus14 for the feedback. Have updated the PR. Should be ready for another review 🙂

**Pull request merged**